Tests/SwiftIfConfigTest/EvaluateTests.swift

```diff
@@ -0,0 +1,149 @@
+      try IfConfigState(condition: condition, configuration: buildConfig)
+    }
+
+    XCTAssertEqual(try ifConfigState("hasFeature(ParameterPacks)"), .active)
+    XCTAssertEqual(try ifConfigState("hasFeature(HigherKindedGenerics)"), .inactive)
+  }
+
+  func testAttributes() throws {
+    let buildConfig = TestingBuildConfiguration(attributes: ["available"])
+
+    func ifConfigState(_ condition: ExprSyntax) throws -> IfConfigState {
+      try IfConfigState(condition: condition, configuration: buildConfig)
+    }
+
+    XCTAssertEqual(try ifConfigState("hasAttribute(available)"), .active)
+    XCTAssertEqual(try ifConfigState("hasAttribute(unsafeUnavailable)"), .inactive)
+  }
+
+  func testPlatform() throws {
+    let buildConfig = TestingBuildConfiguration()
+
+    func ifConfigState(_ condition: ExprSyntax) throws -> IfConfigState {
+      try IfConfigState(condition: condition, configuration: buildConfig)
+    }
+
+    XCTAssertEqual(try ifConfigState("os(Linux)"), .active)
+    XCTAssertEqual(try ifConfigState("os(BeOS)"), .inactive)
+    XCTAssertEqual(try ifConfigState("arch(arm64)"), .active)
+    XCTAssertEqual(try ifConfigState("arch(x86_64)"), .inactive)
+    XCTAssertEqual(try ifConfigState("targetEnvironment(simulator)"), .active)
+    XCTAssertEqual(try ifConfigState("targetEnvironment(blargh)"), .inactive)
+    XCTAssertEqual(try ifConfigState("_endian(little)"), .active)
+    XCTAssertEqual(try ifConfigState("_endian(big)"), .inactive)
+    XCTAssertEqual(try ifConfigState("_runtime(_Native)"), .active)
+    XCTAssertEqual(try ifConfigState("_runtime(_ObjC)"), .inactive)
+    XCTAssertEqual(try ifConfigState("_ptrauth(arm64e)"), .active)
+    XCTAssertEqual(try ifConfigState("_ptrauth(none)"), .inactive)
+    XCTAssertEqual(try ifConfigState("_pointerBitWidth(_64)"), .active)
+    XCTAssertEqual(try ifConfigState("_pointerBitWidth(_32)"), .inactive)
+  }
+
+  func testVersions() throws {
+    let buildConfig = TestingBuildConfiguration()
+
+    func ifConfigState(_ condition: ExprSyntax) throws -> IfConfigState {
+      try IfConfigState(condition: condition, configuration: buildConfig)
+    }
+
+    XCTAssertEqual(try ifConfigState("swift(>=5.5"), .active)
+    XCTAssertEqual(try ifConfigState("swift(<6"), .active)
+    XCTAssertEqual(try ifConfigState("swift(>=6"), .inactive)
+    XCTAssertEqual(try ifConfigState("compiler(>=5.8"), .active)
+    XCTAssertEqual(try ifConfigState("compiler(>=5.9"), .active)
+    XCTAssertEqual(try ifConfigState("compiler(>=5.10"), .inactive)
+    XCTAssertEqual(try ifConfigState(#"_compiler_version("5009.*.1")"#), .active)
+    XCTAssertEqual(try ifConfigState(#"_compiler_version("5009.*.3.2.3")"#), .inactive)
+    XCTAssertEqual(try ifConfigState(#"_compiler_version("5010.*.0")"#), .inactive)
+  }
+
+  func testCanImport() throws {
+    let buildConfig = TestingBuildConfiguration()
+
+    func ifConfigState(_ condition: ExprSyntax) throws -> IfConfigState {
```

Comment:
Why do we have `CanImportExprSyntax` at all? It seems like we should leave it as a normal call expression and let `SwiftIfConfig` deal with it. There are two diagnostics that would have to move into `SwiftIfConfig` (missing major version, wrong number of arguments), but those are straightforward.

Reply:
That's because the diagnostics(wrong label, wrong number of arguments..) are currently generated in `SwiftParserDiagnostics` and a call expression node is not expressive enough to generate them.

Reply:
`CanImportExprSyntax` is gone now